lib/ansible/modules/net_tools/ip.py

```diff
@@ -0,0 +1,472 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2018, Luis Eduardo <leduardo@lsd.ufcg.edu.br>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: ip
```

Comment:
This should be a more specific name that depicts what kind of IP. iproute2_ip_addr?

Reply:
Hi @s-hertel . Thank you for the comment.

The main idea for this module is manage all aspects that iproute2 support. Basically, It supports ```link``` (manage network interfaces, creating, deleting, change mtu status, and others), ```address``` (add, remove, flush IPv4/IPv6 network address), ```route``` , ```arp``` , ```rule``` and ```tunnel``` manipulations. 

The linux CLI implementation to iroute2 is ```ip``` command. I used this name because is more easy remember how to use if you use ip command line. Usage ```iproute2_ip_addr``` as name limits that the module can offer.

Feel free to comment on.